pkg/server/config.go

```diff
@@ -360,6 +362,13 @@ func NewConfig(opts kcpserveroptions.CompletedOptions) (*Config, error) {
 	// Make sure to set our RequestInfoResolver that is capable of populating a RequestInfo even for /services/... URLs.
 	c.GenericConfig.RequestInfoResolver = requestinfo.NewKCPRequestInfoResolver()
 
+	if kcpfeatures.DefaultFeatureGate.Enabled(kcpfeatures.SyncerTunnel) {
+		kubeBasicLongRunningRequestCheck := c.GenericConfig.LongRunningFunc
+		c.GenericConfig.LongRunningFunc = func(r *http.Request, requestInfo *request.RequestInfo) bool {
+			return kubeBasicLongRunningRequestCheck(r, requestInfo) || genericfilters.BasicLongRunningRequestCheck(sets.NewString(""), sets.NewString("tunnel"))(r, requestInfo)
```

Comment:
Sorry, 1 more request: could you please declare the tunnel check function outside and call it from in here, so we aren't allocating new sets for every invocation?

Reply:
done!